### PR TITLE
[core] Update possible reasons for commit conflicts

### PR DESCRIPTION
**Pull request by tsreaper**
### Purpose

We recently found that, if data is written into expired partitions, commit conflicts may occur.

It is not easy to completely work around this scenario, so currently we prefer users to filter out expired data by themselves. This PR updates possible reasons for commit conflicts to remind the users.

### Tests

No tests.

### API and Format

No API changes.

### Documentation

No new features.

### CODE REVIEW

File: paimon-core/src/main/java/org/apache/paimon/operation/FileStoreCommitImpl.java [hunk 1]
```diff
@@ -1130,7 +1129,8 @@ private Pair<RuntimeException, RuntimeException> createConflictException(
                         "3. You're recovering from an old savepoint, or you're creating multiple jobs from a savepoint.",
                         "   The job will fail continuously in this scenario to protect metadata from corruption.",
                         "   You can either recover from the latest savepoint, "
-                                + "or you can revert the table to the snapshot corresponding to the old savepoint.");
+                                + "or you can revert the table to the snapshot corresponding to the old savepoint.",
+                        "4. The parallelism of committer, or the parallelism of compact coordinator (for append only scalable table) is not 1.");
```
Comment (JingsongLi) [inline paimon-core/src/main/java/org/apache/paimon/operation/FileStoreCommitImpl.java:1133]: We don't need this if we have runtime check.

File: paimon-core/src/main/java/org/apache/paimon/operation/FileStoreCommitImpl.java [hunk 2]
```diff
@@ -1115,12 +1116,10 @@ private Pair<RuntimeException, RuntimeException> createConflictException(
                         "Don't panic!",
                         "Conflicts during commits are normal and this failure is intended to resolve the conflicts.",
                         "Conflicts are mainly caused by the following scenarios:",
-                        "1. Your job is suffering from back-pressuring.",
-                        "   There are too many snapshots waiting to be committed "
-                                + "and an exception occurred during the commit procedure "
-                                + "(most probably due to checkpoint timeout).",
-                        "   See https://paimon.apache.org/docs/master/maintenance/write-performance/ "
-                                + "for how to improve writing performance.",
+                        "1. Data is written into expired partitions.",
```
Comment (JingsongLi) [inline paimon-core/src/main/java/org/apache/paimon/operation/FileStoreCommitImpl.java:1119]: Maybe we can check delete files and know it is the true reason?

**Comment by tsreaper** [comment]
Also related to #3340.

**Comment by JingsongLi** [comment]
Fixed in https://github.com/apache/paimon/pull/3945

**Pull request closed without merging**